v2/lockrenewer.go

```diff
@@ -170,3 +179,35 @@ func (plr *peekLockRenewer) startPeriodicRenewal(ctx context.Context, message *a
 		}
 	}
 }
+
+func (plr *peekLockRenewer) renewMessageLock(ctx context.Context, message *azservicebus.ReceivedMessage, options *azservicebus.RenewMessageLockOptions) error {
+	lockLostErr := &azservicebus.Error{Code: azservicebus.CodeLockLost}
+	renewalTimeout := time.Until(*message.LockedUntil) // assumes messages.LockedUntil is set
+	if renewalTimeout < 0 {
+		// if the lock is already expired, we should not attempt to renew it.
+		return lockLostErr
+	}
+	if *plr.renewalTimeout > 0 {
+		renewalTimeout = *plr.renewalTimeout
+	}
+	// we should keep retrying until lock expiry or until message context is done
+	for time.Until(*message.LockedUntil) > 0 && ctx.Err() == nil {
+		err := plr.renewMessageLockWithTimeout(ctx, renewalTimeout, message, options)
+		// exit the renewal if we get any error other than context canceled or if the error is nil.
+		if !errors.Is(err, context.Canceled) {
```

Comment:
I think this is too narrow now.
if the error is lockLost, there is no point retrying for example, and we weren't retrying before, but now you will retry N times and get N lockLost error, instead of exiting promptly and canceling the message context.

I would have just explicitely retried on the deadline exceeded (which is our timeout case).

Reply:
oops, meant to say `errors.Is(err, context.DeadlineExceeded)`.

I think that is what I have? We are only retrying for the timeout case. If the renew fails for any reason other than timeout, we are not retrying

Reply:
added a UT here to show: 
<img width="914" alt="image" src="https://github.com/user-attachments/assets/d36b5d93-a4f2-425f-96c0-7df014a57053">